interfaces/builtin/steam_support.go

```diff
@@ -0,0 +1,201 @@
+  steam-support:
+    deny-connection: true
+    deny-auto-connection: true
+`
+
+const steamSupportConnectedPlugSecComp = `
+# Description: Allow Steam games to work correctly.
+# This extends the browser-support base seccomp. They cannot be used
+# together as browser-support will deny ptrace capabilities.
+
+# This is required for Feral Interactive ports
+ptrace
+
+# Required by titles such as Dying Light
+sched_setaffinity
+`
+
+const steamSupportConnectedPlugAppArmor = `
+# Description: Provide support for the Steam Client and Steam Games to run
+# correctly under full AppArmor confinement. Primary use case is to leverage
+# solus-runtime-gaming and linux-steam-integration to provide a sane and safe
+# Steam experience across all platforms, with appropriate mediation.
+# This AppArmor definition is used in conjunction with the browser-support
+# base AppArmor definition. They cannot both be used together as defined
+# connections however, as browser-support will deny the ptrace capabilities
+# that are required by Steam.
+
+########################
+# Steam Client / API   #
+########################
+
+# Allow Valve IPC to work for the multiprocess client architecture.
+# "Server" component will write/create/lock, clients (i.e. running games in
+# the Steam API) will mmap the object.
+owner /{dev,run}/shm/u*-ValveIPCSharedObj* mrwk,
+
+# Linux Steam Integration config mask system
+owner /{dev,run}/shm/u*-LinuxSteamIntegration.unity3d.* rw,
+
+# i.e. /dev/shm/#539043
+owner /{dev,run}/shm/#* mrw,
+
+# Further required pipe devices (i.e. /dev/shm/u1000-Shm_125376fc)
+owner /{dev,run}/shm/u*-Shm_* mrw,
+
+# Steam Web Store IPC (i.e. /dev/shm/ValveIPCSHM_1000)
+owner /{dev,run}/shm/ValveIPCSHM_* rw,
+
+# For old Source games like L4D2
+owner /{dev,run}/shm/{,.}org.chromium.Chromium.shmem.libcef_* mrwk,
+
+
+########################
+#    Audio Support     #
+########################
+
+# Base runtime can provide HRTF/OpenAL implementation for games
+/usr/share/openal/hrtf/ r,
+/usr/share/openal/hrtf/*.mhr mr,
+/usr/share/openal/presets/ r,
+/usr/share/openal/hrtf/*.{ambdec,txt} mr,
+
+# solus-runtime-gaming uses a stateless alsa configuration which will
+# always use PulseAudio if possible.
+/usr/share/alsa/alsa.conf r,
+/usr/share/defaults/alsa/asound.conf r,
+/usr/share/alsa/alsa.conf.d/ r,
+/usr/share/alsa/alsa.conf.d/** r,
+/usr/share/alsa/cards/** r,
+/usr/share/alsa/pcm/** r,
+
+# PCM Access
+/sys/devices/pci**/sound/**/pcm**/pcm_class r,
+
+########################
+#  Mono Game Support   #
+########################
+
+# i.e. /dev/shm/mono-shared-1000-shared_data-ironhide-Linux-x86_64-328-12-0
+owner /{dev,run}/shm/mono-shared-*-shared_{data,fileshare}-* mrw,
+
+# i.e. /dev/shm/mono.16965
+owner /{dev,run}/shm/mono.@{pid} mrw,
+
+
+########################
+#     GPU Access       #
+########################
+
+# Games can and will attempt to map /dev/nvidia* (specifically Unity)
+/dev/nvidiactl mr,
+/dev/nvidia[0-9] mr,
```

Comment:
'r' is already allowed by the opengl interface. Please remove 'r' and 'plugs: [ opengl ]'. I've seen 'm' elsewhere with nvidia devices. It may make sense to move the 'm'map access to opengl, but for now let's leave 'm' here (we very well may move this to the opengl interface).